src/query/service/src/table_functions/srf/generate_series.rs

```diff
@@ -0,0 +1,381 @@
+                &BUILTIN_FUNCTIONS,
+            )?;
+            let step: F64 = check_number(
+                None,
+                FunctionContext::default(),
+                &Expr::<usize>::Cast {
+                    span: None,
+                    is_try: false,
+                    expr: Box::new(Expr::Constant {
+                        span: None,
+                        scalar: self.step.clone(),
+                        data_type: self.step.clone().as_ref().infer_data_type(),
+                    }),
+                    dest_type,
+                },
+                &BUILTIN_FUNCTIONS,
+            )?;
+
+            self.finished = true;
+            Ok(Some(DataBlock::new_from_columns(vec![
+                Float64Type::from_data(range(start, end, step).into_iter()),
+            ])))
+        } else {
+            let dest_type = DataType::Number(NumberDataType::Int64);
+            let start: i64 = check_number(
+                None,
+                FunctionContext::default(),
+                &Expr::<usize>::Cast {
+                    span: None,
+                    is_try: false,
+                    expr: Box::new(Expr::Constant {
+                        span: None,
+                        scalar: self.start.clone(),
+                        data_type: self.start.clone().as_ref().infer_data_type(),
+                    }),
+                    dest_type: dest_type.clone(),
+                },
+                &BUILTIN_FUNCTIONS,
+            )?;
+            let end: i64 = check_number(
+                None,
+                FunctionContext::default(),
+                &Expr::<usize>::Cast {
+                    span: None,
+                    is_try: false,
+                    expr: Box::new(Expr::Constant {
+                        span: None,
+                        scalar: self.end.clone(),
+                        data_type: self.end.clone().as_ref().infer_data_type(),
+                    }),
+                    dest_type: dest_type.clone(),
+                },
+                &BUILTIN_FUNCTIONS,
+            )?;
+            let step: i64 = check_number(
+                None,
+                FunctionContext::default(),
+                &Expr::<usize>::Cast {
+                    span: None,
+                    is_try: false,
+                    expr: Box::new(Expr::Constant {
+                        span: None,
+                        scalar: self.step.clone(),
+                        data_type: self.step.clone().as_ref().infer_data_type(),
+                    }),
+                    dest_type,
+                },
+                &BUILTIN_FUNCTIONS,
+            )?;
+            self.finished = true;
+            Ok(Some(DataBlock::new_from_columns(vec![
+                Int64Type::from_data(range(start, end, step).into_iter()),
+            ])))
+        }
+    }
+}
+
+pub fn number_scalars_have_float(scalars: Vec<Scalar>) -> bool {
+    for scalar in scalars {
+        match scalar {
```

Comment:
`matches!(...)`

Reply:
How about add a method `is_float` for `Scalar`.

And check `if arg1.is_float() || arg2.is_float() || arg3.is_float()`.